src/main/java/me/redstom/privaterooms/util/room/RoomVisibility.java

```diff
@@ -18,6 +18,61 @@
 
 package me.redstom.privaterooms.util.room;
 
-public enum RoomVisibility {
-    PUBLIC, PRIVATE, HIDDEN
+import me.redstom.privaterooms.util.permission.PermissionSet;
+import net.dv8tion.jda.api.Permission;
+
+import java.util.EnumSet;
+
+public enum RoomVisibility implements PermissionSet {
+    PUBLIC(
+      NONE,
+      NONE,
+      "\uD83D\uDD13"), // "🔓"
+    PRIVATE(
+      NONE,
+      EnumSet.of(Permission.VOICE_CONNECT),
+      "\uD83D\uDD10"), // "🔐"
+    HIDDEN(
+      NONE,
+      EnumSet.of(Permission.VOICE_CONNECT, Permission.VIEW_CHANNEL),
+      "\uD83D\uDD12"); // "🔒"
+
+    private EnumSet<Permission> allow;
+    private EnumSet<Permission> deny;
+    private long allowRaw;
+    private long denyRaw;
+
```

Comment:
Why not using a PermissionSet instead of the 4 fields above?

Reply:
Comment ça ?